### DISCUSSION TITLE
swag (nginx) reverse proxy configuration

### DISCUSSION THREAD

**Discussion by Hackashaq666** [music-assistant/support#2579] [Q&A]
Hi, anyone have a working musicassistant config file for swag (linuxserver nginx) reverse proxy? I am struggling to get this to work with standard template. Any help is appreciated!

```
server {
    listen 443 ssl http2;
    listen [::]:443 ssl http2;

    server_name musicassistant.*;

    include /config/nginx/ssl.conf;

    client_max_body_size 0;

    location / {

        include /config/nginx/proxy.conf;
        include /config/nginx/resolver.conf;
        set $upstream_app 192.168.1.10;
        set $upstream_port 8095;
        set $upstream_proto http;
        proxy_pass $upstream_proto://$upstream_app:$upstream_port;
```

**Comment by GentleHoneyLover** [comment]
Struggling with reverse proxying Music Assistant as well. Anybody had luck with setting it up?

I use Caddy for this purpose in my local network and I'm setting up Music Assistant exactly the same way as a few dozen of my other services but MA doesn't work. I don't know what I'm missing. I run MA in a docker container (and all works fine, web ui can be accessed via local dns record and port or via ip address and port). 

```
*.local.mydomain.com {                         # I use my FQDN instead of mydomain.com, of course ;)
  tls {
    dns cloudflare {env.CLOUDFLARE_TOKEN}
  }

  @musicassistant host ma.local.mydomain.com
  handle @musicassistant {
    reverse_proxy ma.home:8095         # ma.home is resolved by my local dns to 10.0.20.7
  }
}
```

Here's what I see in the Caddy logs (looks like the request to MA is timing out):

`{"level":"error","ts":1732020005.3191288,"logger":"http.log.error","msg":"dial tcp 10.0.20.7:8095: i/o timeout","request":{"remote_ip":"10.0.10.110","remote_port":"50465","client_ip":"10.0.10.110","proto":"HTTP/2.0","method":"GET","host":"ma.local.mydomain.com","uri":"/","headers":{"Sec-Fetch-Dest":["document"],"Accept":["text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"],"Sec-Fetch-Mode":["navigate"],"Accept-Encoding":["gzip, deflate, br"],"Cookie":["REDACTED"],"User-Agent":["Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.1 Safari/605.1.15"],"Sec-Fetch-Site":["none"],"Accept-Language":["en-US,en;q=0.9"],"Priority":["u=0, i"]},"tls":{"resumed":false,"version":772,"cipher_suite":4865,"proto":"h2","server_name":"ma.local.mydomain.com"}},"duration":3.000454938,"status":502,"err_id":"cdtxm7ger","err_trace":"reverseproxy.statusError (reverseproxy.go:1269)"}`

**Reply by GentleHoneyLover** [reply]
Ok, this one was actually solved. It was a networking config problem on my end — had nothing to do with Caddy. The config above is working just fine ;)

**Comment by luclis** [comment]
Mine work while I stay in my local network (probably because DNS isn't resolving outside).
It's pretty standard, hope it may be usefull. 
The h3 referes are for http3 (quic), I don't think it's working on here, I didn't check the conf since I make it work. I could get a look if wanted. 
If someone know how to run all of this in a subfolder (like homeassistant,local/musique), it'ld be great, I don't want to have multiples external dns name. 

```
map $http_upgrade $connection_upgrade {
    default upgrade;
    ''      close;
}

server {
  listen 80;
  #  http2 on; 
  server_name  music-assistant,local;
  return 301 https://$host$request_uri;
  location / {
	  add_header Alt-Svc 'h3=":443"; ma=86400';
  }
}

server {
  listen 443 ssl http2;
  #  listen 443 ssl;
  #  http2 on; 
  #  http3 on; 

  server_name  music-assistant,local;
  ssl_certificate     /etc/ssl/public/music-assistant,local_chain.pem;
  ssl_certificate_key /etc/ssl/private/music-assistant,local.key;

  #Fichier de logs
  access_log      /var/log/nginx/musiqueassistant_access.log;
  error_log       /var/log/nginx/musiqueassistant_error.log;


  location / {
    add_header Strict-Transport-Security "max-age=15552000; includeSubDomains; preload;";
    proxy_pass http://127.0.0.1:8095;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header Host $host;
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $connection_upgrade;
    }
}
```

**Reply by GentleHoneyLover** [reply]
Thanks! It isn't obvious to me which part of your NGINX config does something that my caddy config doesn't :) Is there anything you did on the MASS config side to enable reverse-proxy support?

**Comment by luclis** [comment]
Don't know what it can be... 
Maybe the map part who permit the use of websocket ? 
```
map $http_upgrade $connection_upgrade {
    default upgrade;
    ''      close;
}
```
